service/api_logic/user_logic.py

```diff
@@ -97,7 +97,21 @@ async def sign_up_user(self, email, username, password):
         
         return user
     
+    async def delete_user(self, email):
+        user = self.get_user_by_email_or_username(email = email)
+        if not user:
+            raise UserDoesNotExistError(email)
+        
+        team_type_dto = self.dto_for_type_of_preference(TEAM_TYPE)
```

Comment:
So many function calls without try-except?

I also believe this should all be done in a single transaction and not multiple to avoid cases when we only managed to remove part of the data